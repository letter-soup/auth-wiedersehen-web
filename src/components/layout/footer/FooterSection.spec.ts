import { describe, expect, it, vi } from 'vitest'
import { mount } from '@/lib/tests/mount'
import FooterSection from '@/components/layout/footer/FooterSection.vue'

// Mock LanguageSwitcher component
vi.mock('@/components/primitives/language-switcher/LanguageSwitcher.vue', () => ({
  default: 'language-switcher',
}))

describe('FooterSection (snapshot)', () => {
  it('renders properly', async () => {
    const wrapper = mount(FooterSection)

    expect(wrapper).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
