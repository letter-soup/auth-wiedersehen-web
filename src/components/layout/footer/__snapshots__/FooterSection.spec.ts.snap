// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterSection (snapshot) > renders properly 1`] = `
"<footer class="grid grid-cols-12 gap-2 pb-4">
  <div class="col-start-3 col-span-8 lg:col-start-4 lg:col-span-6 max-sm:col-start-2 max-sm:col-span-10">
    <div class="grid gap-6 grid-cols-3">
      <div class="flex justify-center">
        <div class="w-fit">
          <h2 class="mb-4 text-sm font-semibold text-gray-900 uppercase dark:text-white">footer:resources</h2>
          <ul class="text-gray-600 dark:text-gray-400 space-y-2">
            <li><a class="hover:underline" target="_blank" href="https://github.com/letter-soup/auth-wiedersehen-web">footer:documentation</a></li>
          </ul>
        </div>
      </div>
      <div class="flex justify-center">
        <div class="w-fit">
          <h2 class="mb-4 text-sm font-semibold text-gray-900 uppercase dark:text-white">footer:community</h2>
          <ul class="text-gray-600 dark:text-gray-400 space-y-2">
            <li><a class="hover:underline" target="_blank" href="https://github.com/letter-soup/auth-wiedersehen-web">social-media:github</a></li>
            <li><a class="hover:underline" target="_blank" href="https://discord.gg/dUQvNkZc">social-media:discord</a></li>
          </ul>
        </div>
      </div>
      <div class="flex justify-center">
        <div class="w-fit">
          <h2 class="mb-4 text-sm font-semibold text-gray-900 uppercase dark:text-white">footer:legal</h2>
          <ul class="text-gray-600 dark:text-gray-400 space-y-2">
            <li><a class="hover:underline" target="_blank" href="/terms">footer:terms-of-service</a></li>
            <li><a class="hover:underline" target="_blank" href="/privacy">footer:privacy-policy</a></li>
          </ul>
        </div>
      </div>
    </div>
    <div data-orientation="horizontal" role="separator" class="shrink-0 bg-border relative h-px w-full my-5"></div>
    <div class="flex justify-between items-center"><span class="text-sm text-gray-500 sm:text-center dark:text-gray-400"> © 2025 <a href="/" class="hover:underline">Auth Wiedersehen™</a>. footer:all-rights-reserved</span>
      <language-switcher></language-switcher>
    </div>
  </div>
</footer>"
`;
